### PR TITLE
No libappstream

### PR DESCRIPTION
**Pull request by matthiasclasen**
Parse appdata using GMarkup, and avoid linking libflatpak against libappstream-glib.

### CODE REVIEW

File: common/flatpak-appdata.c [hunk 1]
```diff
@@ -21,7 +21,168 @@
...
+    }
+
+  if (g_str_equal (element_name, "component"))
+    {
+      data->in_component = FALSE;
+    }
+  else if (g_str_equal (element_name, "id"))
+    {
+      if (g_str_equal (text, data->app_id))
+        data->in_component = TRUE;
```
Comment (alexlarsson) [inline common/flatpak-appdata.c]: This seems to rely on "id" being the first (or at least early) element in the component. Maybe a better approach is to always collect everything, including the id, and then return in end of componente if it matches, else clear everything.

Also, we need to see if id == app_id || id == app_id + ".desktop" for backwards compat.

Comment (matthiasclasen) [reply common/flatpak-appdata.c]: Yeah, thats true

**Comment by alexlarsson** [comment]
```
common/flatpak-appdata.c:240:7: runtime error: member access within null pointer of type 'struct GError'
/srv/code/tests/libtest.sh: line 304:  4503 Segmentation fault      (core dumped) ${FLATPAK} ${U} install -y ${REPONAME}-repo org.test.Hello master
```

**Comment by matthiasclasen** [comment]
should work now

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Comment by alexlarsson** [comment]
```
flatpak:ERROR:tests/testlibrary.c:2012:test_transaction_install_uninstall: assertion failed (flatpak_installed_ref_get_appdata_name (ref) == "Hello world test app: org.test.Hello"): (NULL == "Hello world test app: org.test.Hello")
# flatpak-DEBUG: running trigger mime-database.trigger
cleaning up pid 24789
./buildutil/tap-test: line 24: 24767 Aborted                 (core dumped) $WRAPPER ${srcd}/${bn} -k --tap
Bail out! flatpak:ERROR:tests/testlibrary.c:2012:test_transaction_install_uninstall: assertion failed (flatpak_installed_ref_get_appdata_name (ref) == "Hello world test app: org.test.Hello"): (NULL == "Hello world test app: org.test.Hello")
ERROR: testlibrary - Bail out! flatpak:ERROR:tests/testlibrary.c:2012:test_transaction_install_uninstall: assertion failed (flatpak_installed_ref_get_appdata_name (ref) == "Hello world test app: org.test.Hello"): (NULL == "Hello world test app: org.test.Hello")
```

**Comment by alexlarsson** [comment]
Pushed a rebase, lets try this one

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**